hibernate-core/src/main/java/org/hibernate/graph/EffectiveEntityGraph.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.graph;
+
+/**
+ * Encapsulates the entity graph and its "type" (LOAD/FETCH)
+ *
+ * @author Steve Ebersole
+ */
+public class EffectiveEntityGraph {
+	enum GraphType {
```

Comment:
Isn't this now duplicating EntityGraphSemantic?

Reply:
WRT `GraphType` vs `EntityGraphSemantic` yes.  However, I left `EffectiveEntityGraph` in place because I am not yet sure how to best handle this.  IMO `EffectiveEntityGraph` conveys this "pairing" much better that we do currently in `LoadQueryInfluencers`, but using `EffectiveEntityGraph` on `LoadQueryInfluencers` means needing a new instantiation of EffectiveEntityGraph to wrap the EntityGraph every time a new graph is pushed to the `LoadQueryInfluencers`.  So still thinking through this one.